modules/document_repository/jsx/categoryForm.js

```diff
@@ -0,0 +1,211 @@
+        </div>
+      );
+    }
+
+    // Waiting for data to load
+    if (!this.state.isLoaded) {
+      return (
+        <button className="btn-info has-spinner">
+          Loading
+          <span
+            className="glyphicon glyphicon-refresh glyphicon-refresh-animate">
+          </span>
+        </button>
+      );
+    }
+
+    return (
+      <div className="row">
+        <div className="col-md-8 col-lg-7">
+          <FormElement
+            name="docUpload"
+            fileUpload={true}
+            onSubmit={this.handleSubmit}
+            ref="form"
+          >
+            <h3>Add a category</h3><br/>
+            <TextboxElement
+              name="category_name"
+              label="Category Name"
+              onUserInput={this.setFormData}
+              ref="category_name"
+              required={true}
+              value={this.state.formData.category_name}
+            />
+            <SelectElement
+              name="parent_id"
+              label="Parent"
+              options={this.state.data.fieldOptions.fileCategories}
+              onUserInput={this.setFormData}
+              ref="parent_id"
+              hasError={false}
+              value={this.state.formData.parent_id}
+            />
+            <TextareaElement
+              name="comments"
+              label="Comments"
+              onUserInput={this.setFormData}
+              ref="comments"
+              value={this.state.formData.comments}
+            />
+            <ButtonElement label="Add Category"/>
+          </FormElement>
+        </div>
+      </div>
+    );
+  }
+
+/** *******************************************************************************
+ *                      ******     Helper methods     *******
+ *********************************************************************************/
+
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+
+    let formData = this.state.formData;
+    let formRefs = this.refs;
+
+    // Validate the form
+    if (!this.isValidForm(formRefs, formData)) {
+      return;
+    }
+      this.uploadFile();
+  }
+  /*
+   * Uploads the file to the server
+   */
+  uploadFile() {
+    // Set form data and upload the media file
+    let formData = this.state.formData;
+    let formObj = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObj.append(key, formData[key]);
+      }
+    }
+   fetch(this.props.action, {
+      method: 'POST',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+      body: formObj,
+    })
+    .then((resp) => resp.json())
+    .then(()=>{
+        this.props.refreshPage();
+        this.fetchData();
+        // refresh the upload page
+        this.props.newCategoryState();
+        this.setState({
+          formData: {}, // reset form data after successful file upload
+          uploadProgress: -1,
```

Comment:
I believe this line can be deleted since it does not doing anything significant.